src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/PackageActionBaseCommand.cs

```diff
@@ -293,8 +293,23 @@ protected override void EndProcessing()
 
         protected async Task CheckPackageManagementFormat()
         {
-            // check if Project has any packages installed
-            if (!(await Project.GetInstalledPackagesAsync(Token)).Any())
+            bool packagesConfigAndSupportsPackageReferences = false;
+
+            if (Project.ProjectStyle == ProjectModel.ProjectStyle.PackagesConfig)
+            {
+                try
+                {
+                    packagesConfigAndSupportsPackageReferences = Project.ProjectServices.Capabilities.SupportsPackageReferences;
+                }
+                catch (NotImplementedException)
+                {
+                    //If Capabilities aren't implemented, then we assume it doesn't support Package References.
```

Comment:
Not quite true. 
The current implementation is a bit all over, and I think this comment could be interpreted as misleading. 

The reality is the only project services that don't implement the Capabilities interface are the NetCorePackageReference ones and those can only be PackageReference (so they technically do support PR). 

All of this is because the interactions with the projects system change, from pull to push. In packages.config/legacy packagereference we ask the project system, in .NET Core, the project system tells us. 

We have so many different implementations of this check. 
Long term they should probably be united into NuGetProjectUpgradeUtility. 

For now, you can just do a packages.config check alongside with a supportpackagereferences check without the fear of throwing, or at the least remove the comment.

Reply:
`INuGetProjectServices` is a public interface, which means other implementations presumably exist. However, if throwing NotSupported is simply because it's not relevant in NetCore and we had no reason to provide Capabilities. I think we have more of an interface design problem to address. Is that what the follow-up Issue should be about, or what do you think?

I can remove the try/catch, rely on us knowing that NetCore will always be PR, and there's no reason to expect someone has their own implementation that needs to "not break".

Reply:
> if throwing NotSupported is simply because it's not relevant in NetCore and we had no reason to provide Capabilities, then I think we have more of an interface design problem to address

That's why said the current implementation is a bit all over the place :) 
Don't necessarily want into any "interface design" discussion here, but given that we can refactore client code more easily, a consolidated check for upgradeability means fewer bugs ;) 

Referring to my comment above, do one of the 2:

* For now, you can just do a packages.config check alongside with a supportpackagereferences check without the fear of throwing 
* or at the least remove the comment.

Reply:
Removed the try-catch. Reworded some comments.